app/main.py

```diff
@@ -1,2 +1,40 @@
 class Dictionary:
-    pass
+    def __init__(self, initial_capacity: int = 8) -> None:
+        self.capacity = initial_capacity
+        self.size = 0
+        self.table = [None] * self.capacity
+
+    def __setitem__(self, key: object, value: object) -> None:
+        if self.size >= self.capacity * 0.75:
+            self._resize()
+
+        index = hash(key) % self.capacity
+        while self.table[index] is not None:
+            if self.table[index][0] == key:
+                self.table[index] = (key, value)
+                return
+            index = (index + 1) % self.capacity
```

Comment:
The logic for handling hash collisions using linear probing is correct. However, ensure that the task requirements specify linear probing as the collision resolution strategy. If another strategy is required, this logic will need to be adjusted.